### PR TITLE
DOC: Fix regression doc strings

### PR DESCRIPTION
**Pull request by bashtage**
Use pandas tooling to improve and fix OLS/WLS/GLS doc strings
Add stl_decomposition to example gallery

- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
@jbrockmendel This came about by using pandas's validate_docstring.  Should eventually get that wired up (only for warnings for a long while) in CI.

File: statsmodels/regression/linear_model.py [hunk 1]
```diff
@@ -1250,54 +1297,51 @@ def iterative_fit(self, maxiter=3, rtol=1e-4, **kwds):
...
-        X = np.asarray(X, np.float64)
-        _X = X.copy()
+        x = np.asarray(x, np.float64)
+        _x = x.copy()
 
         # the following loops over the first axis,  works for 1d and nd
         for i in range(self.order):
-            _X[(i+1):] = _X[(i+1):] - self.rho[i] * X[0:-(i+1)]
-        return _X[self.order:]
+            _x[(i+1):] = _x[(i+1):] - self.rho[i] * x[0:-(i + 1)]
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: i dont particularly care about `i+1` versus `i + 1`, but it shouldn't be mixed

File: statsmodels/regression/linear_model.py [hunk 2]
```diff
@@ -2044,17 +2165,16 @@ def compare_lr_test(self, restricted, large_sample=False):
         Returns
         -------
         lr_stat : float
-            likelihood ratio, chisquare distributed with df_diff degrees of
-            freedom
+            The likelihood ratio which us chisquare distributed with df_diff
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: us -> is?

File: statsmodels/regression/linear_model.py [hunk 3]
```diff
@@ -2271,35 +2391,36 @@ def get_robustcov_results(self, cov_type='HC1', use_t=None, **kwds):
...
         elif cov_type.upper() in ('HC0', 'HC1', 'HC2', 'HC3'):
-            if kwds:
+            if kwargs:
                 raise ValueError('heteroscedasticity robust covariance '
                                  'does not use keywords')
             res.cov_kwds['description'] = descriptions[cov_type.upper()]
             res.cov_params_default = getattr(self, 'cov_' + cov_type.upper())
         elif cov_type.lower() == 'hac':
-            maxlags = kwds['maxlags']   # required?, default in cov_hac_simple
+            # required?, default in cov_hac_simple
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: "TODO"

File: statsmodels/regression/linear_model.py [hunk 4]
```diff
@@ -2806,60 +2923,56 @@ def el_test(self, b0_vals, param_nums, return_weights=0,
...
+            stochastic regressors are at least as large as non-stochastic
+            regressors.  The default is True.
 
         Returns
         -------
-
-        ci : tuple
-            The confidence interval
+        tuple
+            The confidence interval.
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: for these dont we usually list out the names/types of the tuple entries, e.g.

```
Returns
--------
ci_lower : float
ci_upper : float
```

**Comment by jbrockmendel** [comment]
Couple of comments, otherwise LGTM.

**Pull request merged**